lib/internal/Magento/Framework/MessageQueue/Topology/Config/QueueConfigItem/DataMapper.php

```diff
@@ -59,21 +62,28 @@ public function __construct(
      * Get mapped config data.
      *
      * @return array
+     * @throws LocalizedException
      */
-    public function getMappedData()
+    public function getMappedData(): array
     {
         if (null === $this->mappedData) {
             $this->mappedData = [];
             foreach ($this->configData->get() as $exchange) {
                 $connection = $exchange['connection'];
                 foreach ($exchange['bindings'] as $binding) {
                     if ($binding['destinationType'] === 'queue') {
-                        $queueItems = $this->createQueueItems($binding['destination'], $binding['topic'], $connection);
-                        $this->mappedData = array_merge($this->mappedData, $queueItems);
+                        $queueItems = $this->createQueueItems(
+                            (string) $binding['destination'],
+                            (string) $binding['topic'],
+                            (array) $binding['arguments'],
+                            (string) $connection
+                        );
+                        $this->mappedData += $queueItems;
```

Comment:
Unfortunately it might be not an equal to the array_merge. Could you fix it as described here https://github.com/magento/magento-coding-standard/blob/develop/Magento2/Sniffs/Performance/ForeachArrayMergeSniff.md?